### PR TITLE
[10.0] Add module account_invoice_payment_report

### PR DESCRIPTION
**Pull request by grindtildeath**
Backport v12.0 account invoice report with payments

Replaces https://github.com/OCA/account-invoicing/pull/567

### CODE REVIEW

**Comment by grindtildeath** [comment]
@tschanzt @TDu @jcoux @i-vyshnevska Can you guys please update your review from former PR here ? :hugs:

File: account_invoice_payment_report/views/report_invoice.xml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<odoo>
+
+    <template id="report_invoice_document_with_payments" inherit_id="account.report_invoice_document" primary="True">
+        <xpath expr="//span[@t-field='o.amount_untaxed']/ancestor::table" position="inside">
+            <t t-set="payments_vals" t-value="o._get_payments_vals()"/>
+            <t t-foreach="payments_vals" t-as="payment_vals">
+                <tr>
+                    <td>
+                        <i class="oe_form_field text-right oe_payment_label">Paid on <t t-esc="payment_vals['date']"/></i>
```
Comment (i-vyshnevska) [inline account_invoice_payment_report/views/report_invoice.xml]:
```suggestion
                        <i class="oe_form_field text-right oe_payment_label">Paid on <t t-esc="payment_vals['date']" t-options="{'widget': 'date'}"/></i>
```

File: account_invoice_payment_report/models/account_invoice.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+                amount_to_show, precision_rounding=self.currency_id.rounding
+            ):
+                continue
+            payment_ref = payment.move_id.name
+            if payment.move_id.ref:
+                payment_ref += ' (' + payment.move_id.ref + ')'
+            payment_vals.append(
+                {
+                    'name': payment.name,
+                    'journal_name': payment.journal_id.name,
```
Comment (i-vyshnevska) [inline account_invoice_payment_report/models/account_invoice.py:95]: in the additional table only amount and date  is used, seems some fields are redundant, maybe  cut it to only needed vals and left a hook?

Comment (grindtildeath) [reply account_invoice_payment_report/models/account_invoice.py:95]: This is a backport from Odoo code as stated in the comment at the top of the function, so I prefer to keep it as it is.

**Comment by grindtildeath** [comment]
I guess we can merge this since there were enough approvals on former PR https://github.com/OCA/account-invoicing/pull/567

**Comment by simahawk** [comment]
@grindtildeath can you remove the travis fix as it got merged here? https://github.com/OCA/account-invoice-reporting/pull/217
Also, pls squash the last commits.

**Comment by grindtildeath** [comment]
@simahawk Done thx

**Comment by simahawk** [comment]
> @simahawk Done thx

any reason not to squash? Those commits does not look so important to keep them separated. Especially on a new module.

**Comment by grindtildeath** [comment]
I usually don't squash commits from different authors to respect the contributions of each and every one :wink:

**Review by rafaelbn — APPROVED** [review]
LGTM 👍

**Comment by yvaucher** [comment]
@grindtildeath can you at least either

 squash [cca1252](https://github.com/OCA/account-invoice-reporting/pull/122/commits/cca12522be1d30ea25c4b9370791cf30f666aafe) into [003e084](https://github.com/OCA/account-invoice-reporting/pull/122/commits/003e084c32431d5def0050f3842ad0796e46c6a9)

or reword the commit message.

The auto-commit message of suggestion is not helpful nor descriptive.

**Comment by grindtildeath** [comment]
@yvaucher @simahawk You guys happy now? :stuck_out_tongue_closed_eyes:

**Pull request merged**